pkg/cluster/overridemtu.go

```diff
@@ -0,0 +1,94 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"github.com/coreos/ignition/v2/config/v3_2/types"
+	"github.com/openshift/installer/pkg/asset/ignition"
+	"github.com/openshift/installer/pkg/asset/ignition/bootstrap"
+	"github.com/openshift/installer/pkg/asset/ignition/machine"
+	"github.com/openshift/installer/pkg/asset/machines/machineconfig"
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/ARO-RP/pkg/cluster/graph"
+)
+
+const (
+	IgnFilePath = "/etc/NetworkManager/dispatcher.d/30-eth0-mtu-3900"
+	IgnFileUser = "root"
+	IgnFileMode = 0555
+	IgnFileData = `#!/bin/bash
+
+if [ "$1" == "eth0" ] && [ "$2" == "up" ]; then
+    ip link set $1 mtu 3900
+fi`
+)
+
+func (m *manager) overrideEthernetMTU(g graph.Graph) error {
+	ignFile := ignition.FileFromString(IgnFilePath, IgnFileUser, IgnFileMode, IgnFileData)
+
+	// Extend Ignition config for master nodes
+
+	master := g.Get(&machine.Master{}).(*machine.Master)
+	master.Config.Storage.Files = append(master.Config.Storage.Files, ignFile)
+	data, err := ignition.Marshal(master.Config)
+	if err != nil {
+		return errors.Wrap(err, "failed to Marshal Ignition config")
+	}
+	master.File.Data = data
+
+	// Extend Ignition config for worker nodes by adding a
+	// MachineConfig manifest file to the bootstrap node's
+	// Ignition config.
+
+	ignConfig := types.Config{
+		Ignition: types.Ignition{
+			Version: types.MaxVersion.String(),
+		},
+		Storage: types.Storage{
+			Files: []types.File{ignFile},
+		},
+	}
+
+	rawExt, err := ignition.ConvertToRawExtension(ignConfig)
+	if err != nil {
+		return err
+	}
+
+	configs := []*mcv1.MachineConfig{
+		{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: mcv1.SchemeGroupVersion.String(),
+				Kind:       "MachineConfig",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "99-worker-mtu",
+				Labels: map[string]string{
+					"machineconfiguration.openshift.io/role": "worker",
+				},
+			},
+			Spec: mcv1.MachineConfigSpec{
+				Config: rawExt,
+			},
+		},
+	}
+
+	manifests, err := machineconfig.Manifests(configs, "worker", "/opt/openshift/openshift")
+	if err != nil {
+		return errors.Wrap(err, "failed to create MTU MachineConfig manifest for worker machines")
+	}
+	manifestIgnFile := ignition.FileFromBytes(manifests[0].Filename, "root", 0644, manifests[0].Data)
+
+	bootstrap := g.Get(&bootstrap.Bootstrap{}).(*bootstrap.Bootstrap)
+	bootstrap.Config.Storage.Files = append(bootstrap.Config.Storage.Files, manifestIgnFile)
+	data, err = ignition.Marshal(bootstrap.Config)
+	if err != nil {
+		return errors.Wrap(err, "failed to Marshal Ignition config")
+	}
+	bootstrap.File.Data = data
+
+	return nil
+}
```

Comment:
It looks like something that should be implemented in the installer (our fork first and then upstreamed).

Reply:
I'm not sure what upstreaming this to the installer would even look like.
The installer would have to query for an ARO-specific feature flag, so it seemed more appropriate in the RP.